docs/mpi_functions.rst

```diff
@@ -85,35 +85,37 @@ Here's another trimmed example for an HPC system that uses Slurm as the schedule
 
 
 .. code-block:: python
+    :caption: globus_compute_mpi_function_example.py
 
-   from globus_compute_sdk import MPIFunction
+    from globus_compute_sdk import Executor, MPIFunction
 
-    ep_id = <SPECIFY_ENDPOINT_ID>
+    ep_id = "<SPECIFY_ENDPOINT_ID>"
     func = MPIFunction("hostname")
-    for nodes in range(1,4):
-        executor.resource_specification = {
-             "num_nodes": 2,
-             "ranks_per_node": nodes
-        }
-        fu = executor.submit(func)
-        mpi_result = fu.result()
-        print(mpi_result.stdout)
-
-        # The above line prints:
-        exp-14-08
-        exp-14-20
-
-        exp-14-08
-        exp-14-20
-        exp-14-08
-        exp-14-20
-
-        exp-14-08
-        exp-14-20
-        exp-14-08
-        exp-14-08
-        exp-14-20
-        exp-14-20
+    with Executor(endpoint_id=ep_id) as ex:
+        for nodes in range(1, 4):  # reminder: 1, 2, 3; 4 not included
+            ex.resource_specification = {
+                "num_nodes": 2,
+                "ranks_per_node": nodes
+            }
+            fu = ex.submit(func)
+            mpi_result = fu.result()
+            print(mpi_result.stdout)
+
+    # Expect output similar to:
```

Comment:
Removing the subsequent #'s looks weird, would it be more clear to use triple quotes instead?

```
"""
    # Expect output similar to:
    
    exp-14-08
    exp-14-20
    exp-14-20
    exp-14-20
    ...
"""
```

Reply:
Have done this; how does this look to you?